### PR TITLE
Add weekly CI: provide ONNX TestPyPI packages for verification

### PR DESCRIPTION
**Pull request by jcwchen**
**Description**
Add a weekly (or nightly) CI to automatically produce and upload  TestPyPI packages.

**Motivation**
To enhance robustness, provide a TestPyPI package for partners (e.g., ORT, tesorflow-onnx) to often verify ONNX.

### CODE REVIEW

**Comment by askhade** [comment]
2 comments:
1. Use the release CIs instead of creating a new workflow
2. We need this for all platforms not just windows

**Comment by jcwchen** [comment]
> 2 comments:
> 
> 1. Use the release CIs instead of creating a new workflow
> 2. We need this for all platforms not just windows

Updated. Thanks for the suggestions

File: setup.py [hunk 1]
```diff
@@ -66,8 +67,13 @@
     git_version = None
 
 with open(os.path.join(TOP_DIR, 'VERSION_NUMBER')) as version_file:
+    VERSION_NUMBER = version_file.read().strip()
+    if '--weekly_build' in sys.argv:
```
Comment (askhade) [inline setup.py:71]: where is this arg passed to setup.py... I dont see it in any yamls

File: .github/workflows/release_win.yml [hunk 2]
```diff
@@ -95,4 +99,9 @@ jobs:
     - uses: actions/upload-artifact@v1
       with:
         name: wheels
-        path: ./onnx/dist
\ No newline at end of file
+        path: ./onnx/dist
+
+    - name: Upload wheel to TestPyPI
```
Comment (askhade) [inline .github/workflows/release_win.yml:104]: Looks like this step will run on every CI run. We want to upload only for weekly jobs not for push and pull request runs

File: .github/workflows/release_mac.yml [hunk 3]
```diff
@@ -75,3 +78,8 @@ jobs:
       with:
         name: wheels
         path: dist
+
+    - name: Upload wheel to TestPyPI
```
Comment (askhade) [inline .github/workflows/release_mac.yml:82]: same comment as below for all yamls... this step should only run for weekly job

**Review by askhade — CHANGES_REQUESTED** [review]
comments added

**Comment by jcwchen** [comment]
Updated. I have tested it in my forked repo and onnx-weekly wheel can be produced correctly. This PR is ready for review again. Thanks

**Comment by jcwchen** [comment]
Thanks @prasanthpul for adding TestPyPI credentials

**Review by askhade — APPROVED** [review]

**Pull request merged**